general-concepts/dependencies/text.xml

```diff
@@ -608,6 +619,54 @@ RESTRICT="!test? ( test )"
 DEPEND="test? ( dev-util/foo )"
 </codesample>
 
+</body>
+</section>
+
+<section>
+<title>Circular Dependencies</title>
+<body>
+
+<p>
+Circular dependencies occur if one or more of package's (possibly indirect)
+dependencies depend on the package itself. This creates a dependency cycle where
+each of the packages must technically be installed before the other.
+For example, if package A depends on B, B depends on C and C depends on A, then
+the package manager cannot install A before C, and C before A.
+</p>
+
+<p>
+There are three kinds of circular dependencies:
+</p>
+
+<ol>
+  <li>
+    Weak circular dependencies that occur if only one of the packages strictly
+    needs to be installed before the other. For example, the Python certifi
+    package strictly requires setuptools to build but setuptools requires
+    certifi only for some runtime functionality, permitting it to be installed
+    later. <c>PDEPEND</c> is used to express this and automatically resolve
+    the circular dependency.
+  </li>
+
+  <li>
+    USE-conditional (non-weak) circular dependencies that occur if the cycle
+    applies only to some combination of USE flags on one of the packages.
+    For example, running tests in setuptools requires a number of packages
+    which require setuptools to be installed first. This kind of circular
+    dependency can be resolved by the user by adjusting USE flags on one
+    of the packages, e.g. by disabling tests on setuptools, and reenabling them
+    once the package is initially installed.
+  </li>
+
+  <li>
+    Strong circular dependencies that occur if cycle cannot be resolved using
+    the regular means. For example, CMake used to depend on jsoncpp, while
+    jsoncpp required CMake to build. Resolving this kind of dependency usually
```

Comment:
Again, I'd write `<c>cmake</c>` etc. here. Also, this is about the Gentoo package, whose name is all lowercase (so no `CMake`).

Reply:
I was actually focusing on more generic approach (i.e. you have the same problem if you build straight from sources).